src/main/java/org/opensearch/ad/transport/PreviewAnomalyDetectorTransportAction.java

```diff
@@ -120,6 +126,10 @@ void previewExecute(
         ThreadContext.StoredContext context,
         ActionListener<PreviewAnomalyDetectorResponse> listener
     ) {
+        if (adCircuitBreakerService.isOpen()) {
```

Comment:
Do we track memory usage of preview API?

Reply:
no, we don't currently

Reply:
Not a blocking issue, but I think it can helps if we track memory usage of preview API so we can reject too many preview API calls before circuit breaker triggered.

Reply:
preview API only lasts about seconds.  Getting reserved memory can triggers evicting models in shared cache.  I don't want that disturbance to happen if possible.  How about set a max concurrent preview API run (e.g., 5)?

Reply:
Not sure if preview API can finish in seconds for multi-category HC. I'm ok to limit concurrent preview API run.

Reply:
right, I showed the 2 hours data.  Will add the code in the new commit.

Reply:
added concurrent preview API limit.   Also replaced scriipt in terms aggr with composite aggregation because a runtime more than 1 minute will throw the preview API useless.